src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -335,6 +335,9 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
                 if (x.IsTupleType != y.IsTupleType)
                     return false;
 
+                if (x.IsNativeIntegerType != y.IsNativeIntegerType)
+                    return false;
```

Comment:
without this, we were sometimes thinking that IntPtr and nint were interchangable, when that could actually elad to casting problems.